ambry-cloud/src/main/java/com.github.ambry.cloud/CloudStorageManager.java

```diff
@@ -0,0 +1,119 @@
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.server.ServerErrorCode;
+import com.github.ambry.server.StoreManager;
+import com.github.ambry.store.Store;
+import com.github.ambry.store.StoreException;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The storage manager that does book keeping for all the handled by this vcr node.
+ */
+public class CloudStorageManager implements StoreManager {
+  private final ConcurrentMap<PartitionId, CloudBlobStore> partitionTostore;
+  private final VerifiableProperties properties;
+  private final CloudDestination cloudDestination;
+  private final VcrMetrics vcrMetrics;
+  private final ClusterMap clusterMap;
+  private static final Logger logger = LoggerFactory.getLogger(CloudStorageManager.class);
+
+  public CloudStorageManager(VerifiableProperties properties, VcrMetrics vcrMetrics, CloudDestination cloudDestination,
+      ClusterMap clusterMap) {
+    partitionTostore = new ConcurrentHashMap<>();
+    this.properties = properties;
+    this.cloudDestination = cloudDestination;
+    this.vcrMetrics = vcrMetrics;
+    this.clusterMap = clusterMap;
+  }
+
+  @Override
+  public boolean addBlobStore(ReplicaId replica) {
+    if (partitionTostore.containsKey(replica.getPartitionId())) {
+      return false;
+    }
+    CloudBlobStore cloudStore =
+        new CloudBlobStore(properties, replica.getPartitionId(), cloudDestination, clusterMap, vcrMetrics);
+    partitionTostore.put(replica.getPartitionId(), cloudStore);
+    return startBlobStore(replica.getPartitionId());
+  }
+
+  @Override
+  public boolean shutdownBlobStore(PartitionId id) {
+    CloudBlobStore blobStore = partitionTostore.getOrDefault(id, null);
+    if (blobStore == null) {
+      return false;
+    }
+    blobStore.shutdown();
+    return true;
+  }
+
+  @Override
+  public Store getStore(PartitionId id) {
+    return partitionTostore.getOrDefault(id, null);
+  }
+
+  @Override
+  public boolean scheduleNextForCompaction(PartitionId id) {
+    throw new UnsupportedOperationException("Method not supported");
+  }
+
+  @Override
+  public boolean startBlobStore(PartitionId id) {
+    CloudBlobStore cloudStore = partitionTostore.getOrDefault(id, null);
+    if (cloudStore == null) {
+      return false;
+    }
+    try {
+      cloudStore.start();
+    } catch (StoreException e) {
+      logger.error("Can't start CloudStore " + cloudStore, e);
+      return false;
+    }
+    return true;
+  }
+
+  @Override
+  public boolean controlCompactionForBlobStore(PartitionId id, boolean enabled) {
+    throw new UnsupportedOperationException("Method not supported");
+  }
+
+  @Override
+  public List<PartitionId> setBlobStoreStoppedState(List<PartitionId> partitionIds, boolean markStop) {
```

Comment:
I don't know if the method is required by CloudStorageManager in the future. For now, I feel like it is more related to regular storage manager. How about removing this in interface and ClouldStorageManager? (It's easy to add this method if needed later)

Reply:
There could potentially be some reasons for stopping replication from a vcr. I think it might be worth keeping in the interface.